packages/fxa-payments-server/src/routes/Product/SubscriptionRedirect/index.tsx

```diff
@@ -24,26 +24,36 @@ export const SubscriptionRedirect = ({ plan }: SubscriptionRedirectProps) => {
     downloadURL || productRedirectURLs[product_id] || defaultProductRedirectURL;
 
   useEffect(() => {
-    navigateToUrl(redirectUrl);
-  }, [navigateToUrl, redirectUrl]);
+    window.addEventListener('message', handleIframeTask);
+  });
```

Comment:
I think this `useEffect` could use a few tweaks. Something like this:
```
  useEffect(() => {
    const handleIframeTask = (e: any) => {
      if (e.data === 'submitted survey') {
        navigateToUrl(redirectUrl);
      }
    };
    window.addEventListener('message', handleIframeTask);
    return () => window.removeEventListener('message', handleIframeTask);
  }, [redirectUrl]);
```
This returns a cleanup function to remove the event listener on component un-mount, as well as limiting creation of the `handleIframeTask` function to only when `redirectUrl` changes. Would also be nice to have a more specific parameter type for `(e: any)` - but I'd have to dig around to figure out what that should be.

Reply:
OK I made that change, will be in my next commit.

Reply:
done.